src/main/java/com/microsoft/jenkins/azuread/AzureAdUser.java

```diff
@@ -68,6 +69,14 @@ public static AzureAdUser createFromJwt(JwtClaims claims) throws MalformedClaimE
         if (StringUtils.isEmpty(user.uniqueName)) {
             user.uniqueName = (String) claims.getClaimValue("preferred_username");
         }
+        for (String suffix : unwantedUsernameSuffixes.split(",")) {
+            if (user.uniqueName.endsWith(suffix)) {
+                int idx = user.uniqueName.lastIndexOf(suffix);
+                user.uniqueName = user.uniqueName.substring(0, idx);
```

Comment:
nit: just use substring[0,name.length-suffix.length) to avoid calculate lasteIndex?

Reply:
Yeah, you're right! Fixed